src/Sylius/Behat/Context/Api/Admin/ManagingShippingMethodsContext.php

```diff
@@ -63,6 +65,82 @@ public function iSortShippingMethodsByName(string $sortType = 'ascending'): void
         ]);
     }
 
+    /**
+     * @When I add the :rule rule configured with :weight
+     */
+    public function iAddTheRuleConfiguredWithWeight(string $rule, int $weight): void
+    {
+        $type = StringInflector::nameToLowercaseCode($rule);
+        $this->client->addRequestData('rules', [[
+            'type' => $type,
+            'configuration' => [
+                'weight' => $weight,
+            ],
+        ]]);
+    }
+
+    /**
+     * @When I add the "Total weight greater than or equal" rule configured with invalid data
+     */
+    public function iAddTheTotalWeightGreaterThanOrEqualRuleConfiguredWithInvalidData(): void
+    {
+        $this->client->addRequestData('rules', [
+            [
+                'type' => 'total_weight_greater_than_or_equal',
+                'configuration' => [
+                    'weight' => true,
+                ],
+            ],
+        ]);
+    }
+
+    /**
+     * @When /^I add the "([^"]+)" rule configured with (?:€|£|\$)([^"]+) for ("[^"]+" channel)$/
+     */
+    public function iAddTheRuleConfiguredWithForChannel(string $rule, int $value, ChannelInterface $channel): void
+    {
+        match ($rule) {
```

Comment:
The `match` could match and assign a local variable with the correct `type` and then the `addRequestData` function could be called below